File: ai-classification/src/main/scala/ao/ai/classify/kernel/RandomBinaryClassifier.scala
package ao.ai.classify.kernel

import ao.ai.ml.model.theory.BinaryClassifier
import ao.ai.ml.model.output.BinaryClass
import ao.ai.ml.model.input.RealList
import util.Random

/**
 * User: AO
 * Date: 11/6/10
 * Time: 3:43 AM
 */
object RandomBinaryClassifier extends BinaryClassifier[RealList]
{
  //--------------------------------------------------------------------------
  private val rand = new Random


  //--------------------------------------------------------------------------
  def classify(input: RealList)
      : BinaryClass =
  {
    BinaryClass.create(
      rand.nextBoolean)
  }
}

File: ai-classification/src/main/java/ao/ai/classify/decision/RandomBinaryClassifier.java
package ao.ai.classify.decision;

import ao.ai.ml.model.output.BinaryProbClass;
import ao.ai.ml.model.theory.BinaryProbClassifier;
import ao.util.math.rand.Rand;

/**
 * User: AO
 * Date: Oct 3, 2010
 * Time: 3:43:56 PM
 */
public class RandomBinaryClassifier<T>
        implements BinaryProbClassifier<T>
{
    //------------------------------------------------------------------------
//    private int positiveCount;
//    private int totalCount;


    //------------------------------------------------------------------------
    @Override
    public BinaryProbClass classify(T input)
    {
        return BinaryProbClass.createProb(
                Rand.nextDouble());
    }
}
